src/jsMain/kotlin/io/beatmaps/playlist/playlist.kt

```diff
@@ -315,15 +315,33 @@ val playlistPage = fc<Props> {
                                 draggable(it.map.id, idx) {
                                     attrs.classes = setOf("drag-beatmap")
 
-                                    beatmapInfo {
-                                        obj = it.map
-                                        version = it.map.publishedVersion()
-                                        this.audio = audio
+                                    removeMapPlaylist {
+                                        attrs.obj = it.map
+                                        attrs.audio = audio
+                                        attrs.playlistKey = playlist.playlistId
+                                        attrs.mapId = it.map.id
+                                        attrs.removeMap = {
+                                            setMaps(maps - it)
+                                        }
                                     }
                                 }
                             }
                         }
                     }
+                } else if (playlist?.owner?.id == userData?.userId){
+                    div("playlist") {
+                        maps.map {
+                            removeMapPlaylist {
+                                attrs.obj = it.map
+                                attrs.audio = audio
+                                attrs.playlistKey = playlist!!.playlistId
```

Comment:
Generally best to avoid `!!`
The condition `playlist?.owner?.id == userData?.userId` is actually true if you're logged out and the playlist on the page hasn't loaded yet (or doesn't exist)
In which case this will throw an exception.

It would be better to use `playlist != null && playlist.owner.id == userData?.userId`

Reply:
Actually it probably won't throw an exception because maps would be empty... still

Reply:
I had misunderstood that !! would somehow safely not execute the corresponding area if the null check failed. I'll change it.